src/views/VirtualListClientSize.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import kind from '@enact/core/kind';
+import VirtualListJS, {VirtualListNative} from '@enact/moonstone/VirtualList';
+import qs from 'qs';
+
+const items = [];
+
+// eslint-disable-next-line enact/prop-types
+const renderItem = ({index, ...rest}) => {
+	return (
+		<div {...rest}>
+			{items[index]}
+		</div>
+	);
+};
+
+for (let i = 0; i < 100; i++) {
+	items.push('Item ' + ('00' + i).slice(-3));
+}
+
+const itemSize = 60;
+
+const VirtualListClientSizeView = kind({
+	name: 'VirtualListClientSizeView',
+
+	render: ({location}) => {
+		const search = qs.parse(location.search, {ignoreQueryPrefix: true});
+		const clientSize = search.clientSize ? {clientWidth: 1280, clientHeight: 720} : undefined;
+		const VirtualList = search.type === 'JS' ? VirtualListJS : VirtualListNative;
+
+		return (<div style={{width: '1280px', height: '720px'}}>
+			<VirtualList
+				id="virtualList"
+				clientSize={clientSize}
```

Comment:
I'm not sure if it is too much or not, but...

If passing `undefined` as the `clientSize` value, then `VirtualListCore`'s props has the own property with the `clientSize` property name. So how about the following code? By doing it, we could prevent passing `clientSize` prop with `undefined`.

```
const clientSize = search.clientSize ? {clientSize: {clientWidth: 1280, clientHeight: 720}} : null;
<VirtualList
	{...clientSize}
	id="virtualList"
	dataSize={items.length}
	focusableScrollbar
	itemRenderer={renderItem}
	itemSize={itemSize}
/>
```

Reply:
Good suggestion. But since the goal of this view is for perf testing, I wouldn't really worry about it. Plus, it's easier to read IMHO.

Reply:
I agree with your opinion.